psycopg_pool/psycopg_pool/abc.py

```diff
@@ -28,7 +27,6 @@
 
 # Callbacks to pass the pool to on connection failure
 ConnectFailedCB: TypeAlias = Callable[["ConnectionPool[Any]"], None]
-AsyncConnectFailedCB: TypeAlias = Union[
-    Callable[["AsyncConnectionPool[Any]"], None],
-    Callable[["AsyncConnectionPool[Any]"], Awaitable[None]],
+AsyncConnectFailedCB: TypeAlias = Callable[
+    ["AsyncConnectionPool[Any]"], Optional[Awaitable[None]]
```

Comment:
I don't think that this definition is better than the previous one. I do understand that my comment that `Union` could be dropped in Python 3.8 was wrong (`Callable | Callable` is only supported from Python 3.10), but I would rather keep an `Union` here.

Reply:
Will do...